### ISSUE TITLE
Unfixable SlevomatCodingStandard.Functions.StaticClosure violation

### ISSUE THREAD

**Issue by morozov** [slevomat/coding-standard#1081]
```
$ composer show | grep -P 'coding-standard|php_codesniffer'

doctrine/coding-standard                       8.1.0     The Doctrine Coding Standard is ...
slevomat/coding-standard                       6.3.11    Slevomat Coding Standard for PHP...
squizlabs/php_codesniffer                      3.5.6     PHP_CodeSniffer tokenizes PHP, J...
```
Consider the following code example:

```php
<?php

function test()
{
    yield [
        function (): void {
        },
    ];
}
```
Try fixing:
```
$ phpcbf \
    --standard=Doctrine \
    --sniffs=SlevomatCodingStandard.Functions.StaticClosure,Squiz.Arrays.ArrayDeclaration,Squiz.WhiteSpace.ScopeKeywordSpacing \
    test.php

PHPCBF RESULT SUMMARY
----------------------------------------------------------------------
FILE                                                  FIXED  REMAINING
----------------------------------------------------------------------
/home/morozov/Projects/phpcs-playground/test.php      FAILED TO FIX
----------------------------------------------------------------------
A TOTAL OF -1 ERRORS WERE FIXED IN 1 FILE
----------------------------------------------------------------------
PHPCBF FAILED TO FIX 1 FILE
----------------------------------------------------------------------

Time: 137ms; Memory: 10MB
```

**Comment by morozov** [comment]
I reduced the scope to https://github.com/squizlabs/PHP_CodeSniffer/issues/3060, so maybe it's an invalid issue.

**Comment by kukulich** [comment]
Yes, it looks like problem in PHPCS sniff itself but thank you.
